org.lflang/src/org/lflang/LinguaFranca.xtext

```diff
@@ -134,7 +134,7 @@ TargetDecl:
         ((parens+='(' (init+=Expression (','  init+=Expression)*)? parens+=')')
             | (braces+='{' (init+=Expression (','  init+=Expression)*)? braces+='}')
         )?
-    ) ';'?
+    ) (reset?='reset')? ';'?
```

Comment:
I'm confused about this syntax. What does it mean if a state variable of type `foo*` is marked as `reset`?

Reply:
I also find it a bit strange to see this modifier at the end of the declaration. I think it would make more sense to have it at the beginning.

Reply:
Also, I'm thinking that a keyword like `transient` might be better, because it is a clear adjective and a much less frequently used term in variable names.

Reply:
Isn't `reset` already a keyword (for reset transitions)?

Reply:
I think this needs to have a clear connection with a reset transition.  The state gets reset when a reset transition is taken.

Reply:
The problem with `transient` is that it inverts the modifier and consequently the default (if absent).
Our concept relies on the fact that by default we do not automatically reset a state variable, such that if a reactor was not written with modes in mind we to not leak memory by resetting state variables pointing to allocated memory. The approach with `transient` contradicts this in my opinion.

Reply:
I see why that's desirable. I still think it's better to have `reset state ...` than `state ... reset`.

Do you know what the reset behavior is for state variables with a pointer types?

Reply:
I added the reset as a shorthand notation for resetting the state variable in a reset reaction, so I agree with @edwardalee that it should have a connection to it.

Reply:
In my mind, I do not consider the reset as an integral component of the state variable but rather a handler added to it, like a deadline or an after delay. I also did not want to draw to much attention to it by putting it first but if you think a `reset state` makes more sense, I can change that.

> Do you know what the reset behavior is for state variables with a pointer types?

Well, the variable is reset to the initial value provided by the user. I assumed that there always has to be one but if that was a misconception, I will add an error message in case `reset` is used without an initial value.

Reply:
Handlers go at the end, sure, but this isn't really a handler but a modifier that labels the state as being automatically reset. I suggest we put the `reset` in front of the state variable declaration.

Reply:
Ok